editor/src/components/navigator/left-pane/pages-pane.tsx

```diff
@@ -0,0 +1,236 @@
+
+          if (
+            matchResult[matchResult.length - 1]?.pathname !==
+            matchResult[matchResult.length - 1]?.pathnameBase
+          ) {
+            return [
+              {
+                path: exampleRoute,
+                resolvedPath: exampleRoute,
+                matchesRealRoute: false,
+              },
+            ]
+          }
+
+          return matchResult?.map(
+            (match): RouteMatch => ({
+              resolvedPath: match.pathname,
+              path: '/' + (match.route.path ?? ''),
+              matchesRealRoute: true,
+            }),
+          )
+        }),
+        (l, r) => l?.resolvedPath === r?.resolvedPath,
+      )
+
+      return result
+    },
+    'PagesPane remixRoutes',
+  )
+
+  const [navigationControls] = useAtom(RemixNavigationAtom)
+  const [activeRemixScene] = useAtom(ActiveRemixSceneAtom)
+
+  const pathname = navigationControls[EP.toString(activeRemixScene)]?.location?.pathname ?? ''
+
+  const matchResult = matchRoutes(remixRoutes, pathname)
+
+  if (remixRoutes.length === 0) {
+    return (
+      <FlexColumn
+        style={{
+          height: '100%',
+          overflowY: 'scroll',
+          whiteSpace: 'normal',
+          padding: 16,
+          paddingTop: 64,
+        }}
+      >
+        <Subdued>
+          No featured routes were found in the project. Please add a 'featuredRoutes' field to the
+          'utopia' field in the package.json file.
+        </Subdued>
+      </FlexColumn>
+    )
+  }
+  return (
+    <FlexColumn style={{ height: '100%', overflowY: 'scroll' }}>
+      {remixRoutes.map((route: RouteMatch, index) => {
+        const { path, resolvedPath } = route
+        const pathMatchesActivePath = matchResult?.[0].route.path === path
+        const pathToDisplay = path ?? RemixIndexPathLabel
+
+        return (
+          <PageRouteEntry
+            key={path}
+            routePath={pathToDisplay}
+            resolvedPath={pathMatchesActivePath ? matchResult?.[0].pathname : resolvedPath}
+            active={pathMatchesActivePath}
+            matchesRealRoute={route.matchesRealRoute}
+          />
+        )
+      })}
+    </FlexColumn>
+  )
+})
+
+interface PageRouteEntryProps {
+  routePath: string
+  resolvedPath: string
+  active: boolean
+  matchesRealRoute: boolean
+}
+const PageRouteEntry = React.memo<PageRouteEntryProps>((props) => {
+  const [navigationControls] = useAtom(RemixNavigationAtom)
+  const [activeRemixScene] = useAtom(ActiveRemixSceneAtom)
+
+  const onClick = React.useCallback(() => {
+    void navigationControls[EP.toString(activeRemixScene)]?.navigate(props.resolvedPath)
+  }, [navigationControls, activeRemixScene, props.resolvedPath])
+
+  const resolvedRouteSegments = props.resolvedPath.split('/')
+  const templateRouteSegments = props.routePath.split('/')
+
+  const lastResolvedSegment = resolvedRouteSegments[resolvedRouteSegments.length - 1]
```

Comment:
same question

Reply:
here, I _think_ not

Reply:
.split() always returns an array with >0 length (even if the splitter character is not in the string, it returns the whole string itself), so the index here can not be negative